crates/turborepo-lib/src/commands/link.rs

```diff
@@ -579,4 +585,61 @@ mod test {
             vercel_api_mock::EXPECTED_SPACE_ID
         );
     }
+
+    macro_rules! gitignore_append_tests {
+        ($($name:ident: $value:expr,)*) => {
+        $(
+            #[test]
+            fn $name() {
+                let repo_root = AbsoluteSystemPathBuf::new(
+                    TempDir::new().unwrap().into_path().to_string_lossy(),
+                ).unwrap();
+                let gitignore_path = repo_root.join_component(".gitignore");
+
+                let (input, expected) = $value;
+                let _ = fs::write(&gitignore_path, input).unwrap();
+
+                let _ = add_turbo_to_gitignore(&repo_root);
+
+                let output = fs::read_to_string(&gitignore_path).unwrap();
+                assert_eq!(expected, output);
+            }
+        )*
+        }
+    }
+
+    gitignore_append_tests! {
```

Comment:
This are basically testing that `regex` + `lines` iterator isn't messing with us.

Reply:
Drive-by: you might want to consider using `test_case` instead. Having a unit test entirely in a macro makes it hard to set breakpoints inside of it.

Lots of examples [here](https://github.com/vercel/turbo/blob/d24075fc96d0109b7dc5b016d59ad59774134cfd/crates/turborepo-globwalk/src/lib.rs#L450)